File: Python/Math-Array-String/69.py
# 1689. Partitioning Into Minimum Number Of Deci-Binary Numbers


class Solution:
    def minPartitions(self, n: str) -> int:
       return max(int(digit) for digit in n) 

sol = Solution()
output = sol.minPartitions("32")
print(output) # Output: 3

File: JavaScript/Math-Array-String/69.js
// 1689. Partitioning Into Minimum Number Of Deci-Binary Numbers


/**
 * @param {string} n
 * @return {number}
 */
var minPartitions = function(n) {
    // Find the maximum digit in the string
    let maxDigit = 0;
    for (let char of n) {
        maxDigit = Math.max(maxDigit, parseInt(char));
    }
    return maxDigit;
};

// Examples
console.log(minPartitions("32"));           // Output: 3
console.log(minPartitions("82734"));        // Output: 8
console.log(minPartitions("27346209830709182346")); // Output: 9
